### PR TITLE
[After #238] UD Validation #1 - Local run

### PR DESCRIPTION
**Pull request by sidravi1**
Reviewer: @suzinyou 
Estimate: 1hr

---

## Ticket

Fixes: [AAQ-570](https://idinsight.atlassian.net/browse/AAQ-570)

### Goal

Run validation locally. Returns results like this:

{
  "true_positives": 93,
  "false_negatives": 4,
  "true_negatives": 135,
  "false_positives": 713,
  "precision": 0.11538461538461539,
  "recall": 0.9587628865979382
}

More reporting coming in future PRs!

### Future Tasks (optional)
- Run using GA + AWS Secrets + Data in S3
- Run for different thresholds and return ROC/PR-curve

## How has this been tested?

You can run it locally.

1. Run llm-proxy: `make setup-llm-proxy`
2. Copy files from S3 to your local machine. 
      - S3 URI: s3://aaq-static-resources/validation_data/momconnect_2022/
      - Files: mc_urgency_*.csv
3. Update Makefile in `core_backend/validation/urgency_detection` with paths to your file. Note that paths should be relative to root of repo i.e. `aaq_core`
4. Run `make ud-validation`

NOTE: Suggest starting with the `*_small.csv` files to avoid long runs.

## To-do before merge (optional)

#238 should be merged

## Checklist

Fill with `x` for completed. 

- [x] My code follows the style guidelines of this project
- [x] I have reviewed my own code to ensure good quality
- [x] I have tested the functionality of my code to ensure it works as intended
- [x] I have resolved merge conflicts

(Delete any items below that are not relevant)
- [ ] I have updated the automated tests
- [ ] I have updated the scripts in `scripts/`
- [ ] I have updated the requirements
- [ ] I have updated the README file
- [ ] I have updated affected documentation
- [ ] I have added a blogpost in Latest Updates
- [ ] I have updated the CI/CD scripts in `.github/workflows/`
- [ ] I have updated the Terraform code

[AAQ-570]: https://idinsight.atlassian.net/browse/AAQ-570?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -14,7 +14,7 @@ clean:
 # Note: Run `make fresh-env psycopg2-binary=true` to manually replace psycopg with psycopg2-binary
 fresh-env :
 	conda remove --name $(PROJECT_NAME) --all -y
-	conda create --name $(PROJECT_NAME) python==3.11 -y
+	conda create --name $(PROJECT_NAME) python==3.10 -y
```
Comment (sidravi1) [inline Makefile:17]: We are using 3.10 everywhere else.

File: core_backend/app/auth/routers.py [hunk 2]
```diff
@@ -28,16 +30,16 @@ async def login(form_data: OAuth2PasswordRequestForm = Depends()) -> dict:
             detail="Incorrect username or password",
         )
 
-    return {
-        "access_token": create_access_token(user.username),
-        "token_type": "bearer",
-        "access_level": user.access_level,
-        "username": user.username,
-    }
+    return AuthenticationDetails(
```
Comment (sidravi1) [inline core_backend/app/auth/routers.py:33]: Noticed that the SwaggerUI didn't specify return JSON schema so added it

File: core_backend/validation/urgency_detection/validate_ud.py [hunk 3]
```diff
@@ -0,0 +1,263 @@
...
+        client: AsyncClient,
+        fullaccess_token: str,
+        ud_rules_path: str,
+        ud_rules_col: str,
+        aws_profile: Union[str, None],
+    ) -> AsyncGenerator[None, None]:
+        """Setup UD rules table"""
+        ud_rules_df = pd.read_csv(
+            ud_rules_path,
+        )
```
Comment (suzinyou) [inline core_backend/validation/urgency_detection/validate_ud.py:35]:
```suggestion
        ud_rules_df = pd.read_csv(ud_rules_path)
```

**Review by suzinyou — APPROVED** [review]
Works well! Will raise a PR fixing the validation script as well

**Pull request merged**